lib/ClimaCoreTempestRemap/test/mpi_tests/online_remap.jl

```diff
@@ -0,0 +1,139 @@
+import ClimaCore
+import ClimaCoreTempestRemap as CCTR
+using ClimaComms
+using ClimaCommsMPI
+using MPI
+using ClimaCore: Geometry, Meshes, Domains, Topologies, Spaces, Fields
+using Test
+using Plots
+using ClimaCorePlots
+
+OUTPUT_DIR = mkpath("CI_OUTPUT_DIR")
+
+@testset "distributed online remapping" begin
+    # general setup
+    comms_ctx = ClimaCommsMPI.MPICommsContext()
+    pid, nprocs = ClimaComms.init(comms_ctx)
+
+    comms_ctx_singleton = ClimaComms.SingletonCommsContext()
+
+    # domain
+    radius = 1.0 # unit sphere
+    domain = Domains.SphereDomain(radius)
+
+    # construct source spaces
+    ne_i = 20
+    mesh_i = Meshes.EquiangularCubedSphere(domain, ne_i)
+    topology_i_distr = Topologies.Topology2D(comms_ctx, mesh_i)
+
+    nq_i = 3 # polynomial order for SE discretization
+    quad_i = Spaces.Quadratures.GLL{nq_i}()
+    space_i_distr = Spaces.SpectralElementSpace2D(topology_i_distr, quad_i)
+
+    topology_i_singleton = Topologies.Topology2D(comms_ctx_singleton, mesh_i)
+    space_i_singleton =
+        Spaces.SpectralElementSpace2D(topology_i_singleton, quad_i)
+
+    # construct target spaces
+    ne_o = 5
+    mesh_o = Meshes.EquiangularCubedSphere(domain, ne_o)
+    topology_o_distr = Topologies.Topology2D(comms_ctx, mesh_o)
+
+    nq_o = 4
+    quad_o = Spaces.Quadratures.GLL{nq_o}()
+    space_o_distr = Spaces.SpectralElementSpace2D(topology_o_distr, quad_o)
+
+    topology_o_singleton = Topologies.Topology2D(comms_ctx_singleton, mesh_o)
+    space_o_singleton =
+        Spaces.SpectralElementSpace2D(topology_o_singleton, quad_o)
+
+    # generate test data in the Field format (on non-distributed topo. for non-halo exchange version)
+    field_i_singleton = sind.(Fields.coordinate_field(space_i_singleton).long)
+
+    # global exchange (no buffer fill/send here) - convert to superhalo in next implementation
+    root_pid = 0
+    ClimaComms.gather(comms_ctx, parent(field_i_singleton))
+    field_i_singleton =
+        MPI.bcast(field_i_singleton, root_pid, comms_ctx.mpicomm)
```

Comment:
We should define this in ClimaComms: https://github.com/CliMA/ClimaComms.jl/issues/21